docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+- [Medium Touch](#medium-touch), meaning that there is programmable infrastructure available where we can leverage [provider extensions](../../extensions/README.md#infrastructure-provider) and [`machine-controller-manager`](https://github.com/gardener/machine-controller-manager) in order to manage the network setup and the machines.
+
+The general procedure of bootstrapping an autonomous shoot cluster is similar in both scenarios.
+
+#### High Touch
+
+In this scenario, the starting point is right on the first machine that shall be transformed into a control plane node.
+It is assumed that the network was set up properly (see [Prerequisites](#prerequisites)).
+Since machine management is out of scope, upgrades requiring to roll out the nodes (e.g., Kubernetes version upgrades, operating system upgrades, CA rotation, [see more](../usage/shoot_updates.md#rolling-update-triggers)) must be performed in-place.
+We will leverage the functionality that is to be developed in the scope of [#10219](https://github.com/gardener/gardener/issues/10219).
+This will also include operating system upgrades [planned for Garden Linux](https://github.com/gardener/gardener/issues/10219#issuecomment-2262299606).
+
+#### Medium Touch
+
+In this scenario, we can leverage the existing [`extensions.gardener.cloud/v1alpha1.Infrastructure` API](../extensions/infrastructure.md) in order to let the respective provider extension generate the necessary network resources (VPCs, subnets, etc., whatever is needed for the respective infrastructure).
+`machine-controller-manager` can be used to create the control plane nodes (and later also the worker nodes) of the autonomous shoot cluster.
+
+The challenge is that both Gardener provider extensions and `machine-controller-manager` depend on a pre-existing Kubernetes cluster or in general an API server.
+Hence, we propose to use a [KinD cluster](https://kind.sigs.k8s.io), deploy the needed components into it, and create the necessary `Infrastructure`, `MachineClass`, and `MachineDeployment` resources.
+This way, these components can be reused as-is, without big changes or re-wiring of these complex components to run in-process with a fake client (as we do during the `gardenadm init` flow for the simpler components).
+Once the controllers have reconciled successfully, we can export the created infrastructure state and the `Machine` resources, bootstrap the autonomous shoot cluster similar to the [general flow](#commands) also needed for the [High Touch scenario](#high-touch), and import the state again into the created cluster.
+
+> [!NOTE]
+> Since worker node management will be taken over by `machine-controller-manager`, [step 4](#commands) (joining worker nodes) from above list can be omitted.
+> `machine-controller-manager` will also take over the creation of the control plane nodes, however, bootstrapping and joining them must still be performed by the user (meaning that [step 3](#commands) remains).
```

Comment:
It would be nice if the human user can leverage the user data of the machines to automate nodes bootstrapping and joining in the cluster (probably not possible for the first control plane node).

Reply:
The general idea is that the creation of the VPC and nodes should be automated in the [medium touch scenario](https://github.com/gardener/gardener/blob/68ccf134a903568bb724f0926d5c3dfa2e470379/docs/proposals/28-autonomous-shoot-clusters.md#medium-touch), i.e. the user needs to provide some configuration to determine the amount/type of nodes to be created similar to the worker groups in the `Shoot` specification. Afterwards, the process of creating the first control plane node and joining further nodes is still manual.

There are ideas to automate this further (see https://github.com/gardener/gardener/blob/68ccf134a903568bb724f0926d5c3dfa2e470379/docs/proposals/28-autonomous-shoot-clusters.md#autonomous-shoot-clusters-for-end-users and https://github.com/gardener/gardener/issues/2906). This would be a low or zero touch scenario meaning that almost or no manual action would be required. The current GEP does not focus on this level on automation (see https://github.com/gardener/gardener/blob/68ccf134a903568bb724f0926d5c3dfa2e470379/docs/proposals/28-autonomous-shoot-clusters.md#non-goals). That being said, the work of this GEP should be a useful first step towards achieving this level of automation if desired/required.